In [1]:
# ライブラリのインポート
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
# 元となるデータの取得
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-8/user-8-nonull.csv")
df_origin.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          1947309808          NaN   
1           1                18932          1947328740  Eye Tracker   
2           2                48708          1947358516        Mouse   
3           3                50278          1947360086  Eye Tracker   
4           4                64522          1947374330        Mouse   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-8     Recording9     2025/07/18   
1          JD4  2025/07/18           user-8     Recording9     2025/07/18   
2          JD4  2025/07/18           user-8     Recording9     2025/07/18   
3          JD4  2025/07/18           user-8     Recording9     2025/07/18   
4          JD4  2025/07/18           user-8     Recording9     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN      EyesNotFound   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN      EyesNotFound   
4         2025/07/18  ...                   NaN          Fixation   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                         48.0                     1.0              NaN   
2                         48.0                     1.0              NaN   
3                         48.0                     1.0              NaN   
4                        116.0                     1.0            342.0   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4            107.0                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1               NaN               NaN  
2            1867.0             962.0  
3               NaN               NaN  
4            1863.0             958.0  

[5 rows x 84 columns]

# カラムの整理

In [3]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    10758
Recording timestamp                           10745
Computer timestamp                            10745
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [4]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [5]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-8']
Recording name: ['Recording9']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['17:01:24.322']
Recording start time UTC: ['08:01:24.322']
Recording duration: [194848]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [9.7]
Average calibration accuracy (degrees): [0.94]
Average calibration accuracy (pixels): [34]
Average validation accuracy (mm): [13.3]
Average validation precision SD (mm): [2.5]
Average validation precision RMS (mm): [3.2]
Average validation accuracy (degrees): [1.39]
Average validation precision SD (degrees): [0.27]
Average validation precision RMS (degrees): [0.34]
Average validation accuracy (pixels): [46]
Average validation precision 

## カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-8`|
|Recording name| 録画名|`Recording9`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:01:24.322`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:01:24.322`|
|Recording duration| 録画時間|`194848`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`9.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.94`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`34`pixels|
|Average validation accuracy | 平均バリデーション精度|`13.3`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.39`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.34`degrees|
|Average validation accuracy |平均バリデーション精度| `46`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `11`pixels|

In [6]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          1947309808          NaN   
1           1                18932          1947328740  Eye Tracker   
2           2                48708          1947358516        Mouse   
3           3                50278          1947360086  Eye Tracker   
4           4                64522          1947374330        Mouse   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1          1.962609e+09             NaN         NaN           NaN   
2                   NaN             NaN         NaN           NaN   
3          1.962640e+09             NaN         NaN           NaN   
4                   NaN             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3           NaN                NaN  ...                    NaN   
4           NaN                NaN  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1       EyesNotFound                         48.0                      1.0   
2       EyesNotFound                         48.0                      1.0   
3       EyesNotFound                         48.0                      1.0   
4           Fixation                        116.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4             342.0             107.0                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN               NaN               NaN  
2                         NaN            1867.0             962.0  
3                         NaN               NaN               NaN  
4                         NaN            1863.0             958.0  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [7]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             10758
Recording timestamp                    10745
Computer timestamp                     10745
Sensor                                     3
Eyetracker timestamp                   10549
Event                                      9
Event value                                9
Gaze point X                            1631
Gaze point Y                            1017
Gaze point left X                       1623
Gaze point left Y                       1036
Gaze point right X                      1590
Gaze point right Y                       960
Gaze direction left X                   7919
Gaze direction left Y                   7705
Gaze direction left Z                   5518
Gaze direction right X                  6573
Gaze direction right Y                  6402
Gaze direction right Z                  4791
Pupil diameter left                      777
Pupil diameter right                     820
Pupil diameter filtered                  780
Validity l

## Sensor事にデータフレームを分割

In [8]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の既知代わりの時間を取得する

In [9]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0          1947309808    NaN   
238           238              2931524          1950241332    NaN   
398           398              4534933          1951844741    NaN   
406           406              4638911          1951948719    NaN   
1698         1698             43425377          1990735185    NaN   
1704         1704             43567412          1990877220    NaN   
1721         1721             44014159          1991323967    NaN   
1722         1722             44014159          1991323967    NaN   
2024         2024             49021032          1996330840    NaN   
2025         2025             49021032          1996330840    NaN   
3959         3959             81198745          2028508553    NaN   
3963         3963             81237952          2028547760    NaN   
3964         3964             81237952          2028547760    NaN   
4264         4264             86228513          2033538321    NaN   
4265         4265             86228513          2033538321    NaN   
5833         5833            112575380          2059885188    NaN   
5835         5835            112603231          2059913039    NaN   
5836         5836            112603231          2059913039    NaN   
6138         6138            117611023          2064920831    NaN   
6139         6139            117611023          2064920831    NaN   
8108         8108            150349885          2097659693    NaN   
8110         8110            150378044          2097687852    NaN   
8111         8111            150378044          2097687852    NaN   
8403         8403            155385662          2102695470    NaN   
8404         8404            155385662          2102695470    NaN   
10462       10462            189625117          2136934925    NaN   
10465       10465            189654192          2136964000    NaN   
10466       10466            189654192          2136964000    NaN   
10754       10754            194662068          2141971876    NaN   
10757       10757            194847793          2142157601    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
238                     NaN  Eye tracker Calibration start         NaN   
398                     NaN                     MouseEvent  Down, Left   
406                     NaN                     MouseEvent    Up, Left   
1698                    NaN                     MouseEvent  Down, Left   
1704                    NaN                     MouseEvent    Up, Left   
1721                    NaN    Eye tracker Calibration end         NaN   
1722                    NaN             ImageStimulusStart      blank2   
2024                    NaN               ImageStimulusEnd      blank2   
2025                    NaN             ImageStimulusStart  視線誘導問題.002   
3959                    NaN                  KeyboardEvent       Space   
3963                    NaN               ImageStimulusEnd  視線誘導問題.002   
3964                    NaN             ImageStimulusStart      blank2   
4264                    NaN               ImageStimulusEnd      blank2   
4265                    NaN             ImageStimulusStart  視線誘導問題.001   
5833                    NaN                  KeyboardEvent       Space   
5835                    NaN               ImageStimulusEnd  視線誘導問題.001   
5836                    NaN             ImageStimulusStart      blank2   
6138                    NaN               ImageStimulusEnd      blank2   
6139                    NaN             ImageStimulusStart  視線誘導問題.004   
8108                    NaN                  KeyboardEvent       Space   
8110                    NaN               ImageStimulusEnd  視線誘導問題.004   
8111                    NaN             ImageStimulusStart      blank2   
8403                    NaN               ImageStimulusEnd      blank2   
8404             

In [10]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
2025              49021032          1996330840  視線誘導問題.002
3963              81237952          2028547760  視線誘導問題.002
4265              86228513          2033538321  視線誘導問題.001
5835             112603231          2059913039  視線誘導問題.001
6139             117611023          2064920831  視線誘導問題.004
8110             150378044          2097687852  視線誘導問題.004
8404             155385662          2102695470  視線誘導問題.003
10465            189654192          2136964000  視線誘導問題.003


### 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|2025|49021032|1996330840|視線誘導問題.002|
|3963|81237952|2028547760|視線誘導問題.002|
|4265|86228513|2033538321|視線誘導問題.001|
|5835|112603231|2059913039|視線誘導問題.001|
|6139|117611023|2064920831|視線誘導問題.004|
|8110|150378044|2097687852|視線誘導問題.004|
|8404|155385662|2102695470|視線誘導問題.003|
|10465|189654192|2136964000|視線誘導問題.003|

In [11]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-8/start-stop-timing.csv")

In [12]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
2              2                48708          1947358516  Mouse   
4              4                64522          1947374330  Mouse   
6              6                80518          1947390326  Mouse   
8              8                96517          1947406325  Mouse   
10            10               112493          1947422301  Mouse   
...          ...                  ...                 ...    ...   
1712        1712             43843267          1991153075  Mouse   
1713        1713             43859257          1991169065  Mouse   
1715        1715             43973005          1991282813  Mouse   
1718        1718             43995649          1991305457  Mouse   
1719        1719             44003462          1991313270  Mouse   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2                      NaN   NaN         NaN           NaN           NaN   
4                      NaN   NaN         NaN           NaN           NaN   
6                      NaN   NaN         NaN           NaN           NaN   
8                      NaN   NaN         NaN           NaN           NaN   
10                     NaN   NaN         NaN           NaN           NaN   
...                    ...   ...         ...           ...           ...   
1712                   NaN   NaN         NaN           NaN           NaN   
1713                   NaN   NaN         NaN           NaN           NaN   
1715                   NaN   NaN         NaN           NaN           NaN   
1718                   NaN   NaN         NaN           NaN           NaN   
1719                   NaN   NaN         NaN           NaN           NaN   

      Gaze point left X  ...  Original Media height  Eye movement type  \
2                   NaN  ...                    NaN       EyesNotFound   
4                   NaN  ...                    NaN           Fixation   
6                   NaN  ...                    NaN           Fixation   
8                   NaN  ...                    NaN           Fixation   
10                  NaN  ...                    NaN           Fixation   
...                 ...  ...                    ...                ...   
1712                NaN  ...                    NaN                NaN   
1713                NaN  ...                    NaN                NaN   
1715                NaN  ...                    NaN            Saccade   
1718                NaN  ...                    NaN            Saccade   
1719                NaN  ...                    NaN            Saccade   

      Eye movement event duration  Eye movement type index  Fixation point X  \
2                            48.0                      1.0               NaN   
4                           116.0                      1.0             342.0   
6                           116.0                      1.0             342.0   
8                           116.0                      1.0             342.0   
10                          116.0                      1.0             342.0   
...                           ...                      ...               ...   
1712                          NaN                      NaN               NaN   
1713                          NaN                      NaN               NaN   
1715                        100.0                     74.0               NaN   
1718                        100.0                     74.0               NaN   
1719                        100.0                     74.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
2                  NaN                         NaN   
4                107.0                         NaN   
6                107.0                         NaN   
8                107.0                         NaN   
10               107.0                         NaN   
...                ...                         ...   
1712               NaN                         N

In [13]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
1               1                18932          1947328740  Eye Tracker   
3               3                50278          1947360086  Eye Tracker   
5               5                66921          1947376729  Eye Tracker   
7               7                83512          1947393320  Eye Tracker   
9               9               100406          1947410214  Eye Tracker   
...           ...                  ...                 ...          ...   
10751       10751            194617308          2141927116  Eye Tracker   
10752       10752            194633535          2141943343  Eye Tracker   
10753       10753            194650716          2141960524  Eye Tracker   
10755       10755            194667324          2141977132  Eye Tracker   
10756       10756            194683459          2141993267  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1              1.962609e+09   NaN         NaN           NaN           NaN   
3              1.962640e+09   NaN         NaN           NaN           NaN   
5              1.962657e+09   NaN         NaN         334.0          91.0   
7              1.962673e+09   NaN         NaN         347.0         103.0   
9              1.962690e+09   NaN         NaN         208.0         194.0   
...                     ...   ...         ...           ...           ...   
10751          2.157195e+09   NaN         NaN         512.0         528.0   
10752          2.157211e+09   NaN         NaN         563.0         493.0   
10753          2.157229e+09   NaN         NaN         606.0         503.0   
10755          2.157245e+09   NaN         NaN         599.0         509.0   
10756          2.157261e+09   NaN         NaN         581.0         516.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
3                    NaN  ...                    NaN       EyesNotFound   
5                  483.0  ...                    NaN           Fixation   
7                  487.0  ...                    NaN           Fixation   
9                    NaN  ...                    NaN           Fixation   
...                  ...  ...                    ...                ...   
10751              569.0  ...                 1080.0       Unclassified   
10752              563.0  ...                 1080.0       Unclassified   
10753              606.0  ...                 1080.0            Saccade   
10755              597.0  ...                    NaN            Saccade   
10756              565.0  ...                    NaN       Unclassified   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                             48.0                      1.0               NaN   
3                             48.0                      1.0               NaN   
5                            116.0                      1.0             342.0   
7                            116.0                      1.0             342.0   
9                            116.0                      1.0             342.0   
...                            ...                      ...               ...   
10751                         33.0                    425.0               NaN   
10752                         33.0                    425.0               NaN   
10753                         33.0                    974.0               NaN   
10755                         33.0                    974.0               NaN   
10756                         16.0                    426.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
3                   NaN                         NaN   
5                 107.0                         NaN   
7                 107.0                         NaN   
9                 107.0         

## Mouse以外のデータを抽出

In [14]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [15]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [16]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             10578
Recording timestamp                    10569
Computer timestamp                     10569
Sensor                                     2
Eyetracker timestamp                   10549
Event                                      9
Event value                                9
Gaze point X                            1631
Gaze point Y                            1017
Gaze point left X                       1623
Gaze point left Y                       1036
Gaze point right X                      1590
Gaze point right Y                       960
Gaze direction left X                   7919
Gaze direction left Y                   7705
Gaze direction left Z                   5518
Gaze direction right X                  6573
Gaze direction right Y                  6402
Gaze direction right Z                  4791
Pupil diameter left                      777
Pupil diameter right                     820
Pupil diameter filtered                  780
Validity l

In [17]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          1947309808          NaN   
1           1                18932          1947328740  Eye Tracker   
3           3                50278          1947360086  Eye Tracker   
5           5                66921          1947376729  Eye Tracker   
7           7                83512          1947393320  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1          1.962609e+09             NaN         NaN           NaN   
3          1.962640e+09             NaN         NaN           NaN   
5          1.962657e+09             NaN         NaN         334.0   
7          1.962673e+09             NaN         NaN         347.0   

   Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0           NaN                NaN  ...                                  NaN   
1           NaN                NaN  ...                                  NaN   
3           NaN                NaN  ...                                  NaN   
5          91.0              483.0  ...                                  NaN   
7         103.0              487.0  ...                                  NaN   

   Original Media width  Original Media height  Eye movement type  \
0                   NaN                    NaN                NaN   
1                   NaN                    NaN       EyesNotFound   
3                   NaN                    NaN       EyesNotFound   
5                   NaN                    NaN           Fixation   
7                   NaN                    NaN           Fixation   

   Eye movement event duration  Eye movement type index  Fixation point X  \
0                          NaN                      NaN               NaN   
1                         48.0                      1.0               NaN   
3                         48.0                      1.0               NaN   
5                        116.0                      1.0             342.0   
7                        116.0                      1.0             342.0   

   Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0               NaN                         NaN                         NaN  
1               NaN                         NaN                         NaN  
3               NaN                         NaN                         NaN  
5             107.0                         NaN                         NaN  
7             107.0                         NaN                         NaN  

[5 rows x 55 columns]

In [18]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [19]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
1               1                18932          1947328740  Eye Tracker   
3               3                50278          1947360086  Eye Tracker   
5               5                66921          1947376729  Eye Tracker   
7               7                83512          1947393320  Eye Tracker   
9               9               100406          1947410214  Eye Tracker   
...           ...                  ...                 ...          ...   
10751       10751            194617308          2141927116  Eye Tracker   
10752       10752            194633535          2141943343  Eye Tracker   
10753       10753            194650716          2141960524  Eye Tracker   
10755       10755            194667324          2141977132  Eye Tracker   
10756       10756            194683459          2141993267  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1              1.962609e+09   NaN         NaN           NaN           NaN   
3              1.962640e+09   NaN         NaN           NaN           NaN   
5              1.962657e+09   NaN         NaN         334.0          91.0   
7              1.962673e+09   NaN         NaN         347.0         103.0   
9              1.962690e+09   NaN         NaN         208.0         194.0   
...                     ...   ...         ...           ...           ...   
10751          2.157195e+09   NaN         NaN         512.0         528.0   
10752          2.157211e+09   NaN         NaN         563.0         493.0   
10753          2.157229e+09   NaN         NaN         606.0         503.0   
10755          2.157245e+09   NaN         NaN         599.0         509.0   
10756          2.157261e+09   NaN         NaN         581.0         516.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
3                    NaN  ...                    NaN       EyesNotFound   
5                  483.0  ...                    NaN           Fixation   
7                  487.0  ...                    NaN           Fixation   
9                    NaN  ...                    NaN           Fixation   
...                  ...  ...                    ...                ...   
10751              569.0  ...                 1080.0       Unclassified   
10752              563.0  ...                 1080.0       Unclassified   
10753              606.0  ...                 1080.0            Saccade   
10755              597.0  ...                    NaN            Saccade   
10756              565.0  ...                    NaN       Unclassified   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                             48.0                      1.0               NaN   
3                             48.0                      1.0               NaN   
5                            116.0                      1.0             342.0   
7                            116.0                      1.0             342.0   
9                            116.0                      1.0             342.0   
...                            ...                      ...               ...   
10751                         33.0                    425.0               NaN   
10752                         33.0                    425.0               NaN   
10753                         33.0                    974.0               NaN   
10755                         33.0                    974.0               NaN   
10756                         16.0                    426.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
3                   NaN                         NaN   
5                 107.0                         NaN   
7                 107.0                         NaN   
9                 107.0         

## 各問題毎のデータへ切り分け

In [20]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [21]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2026        2026             49036400          1996346208  Eye Tracker   
2027        2027             49052718          1996362526  Eye Tracker   
2028        2028             49069451          1996379259  Eye Tracker   
2029        2029             49086317          1996396125  Eye Tracker   
2030        2030             49102475          1996412283  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2026          2.011623e+09   NaN         NaN         631.0         477.0   
2027          2.011640e+09   NaN         NaN           NaN           NaN   
2028          2.011656e+09   NaN         NaN         620.0         481.0   
2029          2.011673e+09   NaN         NaN         654.0         473.0   
2030          2.011689e+09   NaN         NaN           NaN           NaN   

      Gaze point left X  ...  Original Media height  Eye movement type  \
2026              583.0  ...                 1080.0       Unclassified   
2027                NaN  ...                 1080.0       EyesNotFound   
2028              620.0  ...                 1080.0            Saccade   
2029              630.0  ...                 1080.0            Saccade   
2030                NaN  ...                 1080.0       EyesNotFound   

      Eye movement event duration  Eye movement type index  Fixation point X  \
2026                         33.0                     21.0               NaN   
2027                         17.0                     20.0               NaN   
2028                         33.0                     89.0               NaN   
2029                         33.0                     89.0               NaN   
2030                         17.0                     21.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
2026               NaN                         NaN   
2027               NaN                         NaN   
2028               NaN                         NaN   
2029               NaN                         NaN   
2030               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
2026                         NaN               NaN               NaN  
2027                         NaN               NaN               NaN  
2028                         NaN               NaN               NaN  
2029                         NaN               NaN               NaN  
2030                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [22]:
print(extracted_data["視線誘導問題.002"].nunique(dropna=False).to_string())

Unnamed: 0                             1936
Recording timestamp                    1936
Computer timestamp                     1936
Sensor                                    1
Eyetracker timestamp                   1936
Event                                     1
Event value                               1
Gaze point X                            887
Gaze point Y                            644
Gaze point left X                       851
Gaze point left Y                       644
Gaze point right X                      737
Gaze point right Y                      504
Gaze direction left X                  1670
Gaze direction left Y                  1645
Gaze direction left Z                  1551
Gaze direction right X                 1164
Gaze direction right Y                 1142
Gaze direction right Z                 1066
Pupil diameter left                     574
Pupil diameter right                    521
Pupil diameter filtered                 555
Validity left                   

In [23]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
4266        4266             86228637          2033538445  Eye Tracker   
4267        4267             86245853          2033555661  Eye Tracker   
4268        4268             86261712          2033571520  Eye Tracker   
4269        4269             86278652          2033588460  Eye Tracker   
4270        4270             86295776          2033605584  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
4266          2.048813e+09   NaN         NaN         931.0         303.0   
4267          2.048830e+09   NaN         NaN         930.0         296.0   
4268          2.048846e+09   NaN         NaN         932.0         299.0   
4269          2.048863e+09   NaN         NaN         932.0         297.0   
4270          2.048880e+09   NaN         NaN         936.0         298.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
4266              920.0  ...                 1080.0           Fixation   
4267              911.0  ...                 1080.0           Fixation   
4268              902.0  ...                 1080.0           Fixation   
4269              910.0  ...                 1080.0           Fixation   
4270              919.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
4266                        566.0                    197.0             935.0   
4267                        566.0                    197.0             935.0   
4268                        566.0                    197.0             935.0   
4269                        566.0                    197.0             935.0   
4270                        566.0                    197.0             935.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
4266             297.0                      0.4868   
4267             297.0                      0.4868   
4268             297.0                      0.4868   
4269             297.0                      0.4868   
4270             297.0                      0.4868   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
4266                      0.2752               NaN               NaN  
4267                      0.2752               NaN               NaN  
4268                      0.2752               NaN               NaN  
4269                      0.2752               NaN               NaN  
4270                      0.2752               NaN               NaN  

[5 rows x 57 columns]

In [24]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
8405        8405            155398157          2102707965  Eye Tracker   
8406        8406            155415024          2102724832  Eye Tracker   
8407        8407            155432185          2102741993  Eye Tracker   
8408        8408            155448127          2102757935  Eye Tracker   
8409        8409            155464942          2102774750  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
8405          2.117978e+09   NaN         NaN         944.0         617.0   
8406          2.117995e+09   NaN         NaN         930.0         607.0   
8407          2.118012e+09   NaN         NaN         937.0         602.0   
8408          2.118028e+09   NaN         NaN         933.0         604.0   
8409          2.118045e+09   NaN         NaN         941.0         607.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
8405              954.0  ...                 1080.0           Fixation   
8406              929.0  ...                 1080.0           Fixation   
8407              937.0  ...                 1080.0           Fixation   
8408              935.0  ...                 1080.0           Fixation   
8409              938.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
8405                        216.0                    432.0             945.0   
8406                        216.0                    432.0             945.0   
8407                        216.0                    432.0             945.0   
8408                        216.0                    432.0             945.0   
8409                        216.0                    432.0             945.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
8405             608.0                      0.4924   
8406             608.0                      0.4924   
8407             608.0                      0.4924   
8408             608.0                      0.4924   
8409             608.0                      0.4924   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
8405                      0.5629               NaN               NaN  
8406                      0.5629               NaN               NaN  
8407                      0.5629               NaN               NaN  
8408                      0.5629               NaN               NaN  
8409                      0.5629               NaN               NaN  

[5 rows x 57 columns]

In [25]:
extracted_data["視線誘導問題.004"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
6140        6140            117612728          2064922536  Eye Tracker   
6141        6141            117629175          2064938983  Eye Tracker   
6142        6142            117645923          2064955731  Eye Tracker   
6143        6143            117662526          2064972334  Eye Tracker   
6144        6144            117679048          2064988856  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
6140          2.080195e+09   NaN         NaN           NaN           NaN   
6141          2.080212e+09   NaN         NaN         955.0         288.0   
6142          2.080229e+09   NaN         NaN         840.0         272.0   
6143          2.080245e+09   NaN         NaN           NaN           NaN   
6144          2.080262e+09   NaN         NaN         673.0         231.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
6140                NaN  ...                 1080.0       EyesNotFound   
6141              910.0  ...                 1080.0            Saccade   
6142              812.0  ...                 1080.0            Saccade   
6143                NaN  ...                 1080.0       EyesNotFound   
6144              638.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
6140                         17.0                    256.0               NaN   
6141                         33.0                    462.0               NaN   
6142                         33.0                    462.0               NaN   
6143                         17.0                    257.0               NaN   
6144                        233.0                    312.0             672.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
6140               NaN                         NaN   
6141               NaN                         NaN   
6142               NaN                         NaN   
6143               NaN                         NaN   
6144             230.0                      0.3502   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
6140                         NaN               NaN               NaN  
6141                         NaN               NaN               NaN  
6142                         NaN               NaN               NaN  
6143                         NaN               NaN               NaN  
6144                      0.2132               NaN               NaN  

[5 rows x 57 columns]

In [26]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.002
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.002']
  Presented Media name: ['視線誘導問題.002.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.001
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.001']
  Presented Media name: ['視線誘導問題.001.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.004
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.004']
  Presen

In [27]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.002: Removed 13 columns -> Remaining 44 columns
視線誘導問題.001: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns
視線誘導問題.003: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-8`|
|Recording name| 録画名|`Recording9`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:01:24.322`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:01:24.322`|
|Recording duration| 録画時間|`194848`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`9.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.94`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`34`pixels|
|Average validation accuracy | 平均バリデーション精度|`13.3`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.39`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.34`degrees|
|Average validation accuracy |平均バリデーション精度| `46`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `11`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [28]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
4266        4266             86228637          2033538445   
4267        4267             86245853          2033555661   
4268        4268             86261712          2033571520   
4269        4269             86278652          2033588460   
4270        4270             86295776          2033605584   
...          ...                  ...                 ...   
5829        5829            112520861          2059830669   
5830        5830            112537051          2059846859   
5831        5831            112553946          2059863754   
5832        5832            112570531          2059880339   
5834        5834            112587021          2059896829   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
4266          2.048813e+09         931.0         303.0              920.0   
4267          2.048830e+09         930.0         296.0              911.0   
4268          2.048846e+09         932.0         299.0              902.0   
4269          2.048863e+09         932.0         297.0              910.0   
4270          2.048880e+09         936.0         298.0              919.0   
...                    ...           ...           ...                ...   
5829          2.075104e+09           NaN           NaN                NaN   
5830          2.075120e+09        1059.0         247.0             1034.0   
5831          2.075137e+09        1056.0         245.0             1030.0   
5832          2.075153e+09           NaN           NaN                NaN   
5834          2.075170e+09        1064.0         250.0             1032.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
4266              304.0               942.0               301.0  ...   
4267              298.0               950.0               295.0  ...   
4268              299.0               962.0               299.0  ...   
4269              295.0               953.0               299.0  ...   
4270              300.0               954.0               297.0  ...   
...                 ...                 ...                 ...  ...   
5829                NaN                 NaN                 NaN  ...   
5830              243.0              1083.0               252.0  ...   
5831              238.0              1083.0               251.0  ...   
5832                NaN                 NaN                 NaN  ...   
5834              246.0              1096.0               254.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
4266                       0.2817                        0.4907   
4267                       0.2756                        0.4946   
4268                       0.2765                        0.5010   
4269                       0.2735                        0.4965   
4270                       0.2776                        0.4967   
...                           ...                           ...   
5829                          NaN                           NaN   
5830                       0.2246                        0.5640   
5831                       0.2204                        0.5640   
5832                          NaN                           NaN   
5834                       0.2276                        0.5707   

      Gaze point right Y (MCSnorm)  Eye movement type  \
4266                        0.2788           Fixation   
4267                        0.2734           Fixation   
4268                        0.2765           Fixation   
4269                        0.2765           Fixation   
4270                        0.2751           Fixation   
...                            ...                ...   
5829                           NaN       EyesNotFound   
5830                        0.2337           Fixation   
5831                        0.2327           Fixation   
5832                           NaN           Fixation   
5834                        0.2347           Fixati

In [29]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-8/user-8-pic-1.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-8`|
|Recording name| 録画名|`Recording9`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:01:24.322`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:01:24.322`|
|Recording duration| 録画時間|`194848`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`9.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.94`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`34`pixels|
|Average validation accuracy | 平均バリデーション精度|`13.3`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.39`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.34`degrees|
|Average validation accuracy |平均バリデーション精度| `46`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `11`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [30]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
6140        6140            117612728          2064922536   
6141        6141            117629175          2064938983   
6142        6142            117645923          2064955731   
6143        6143            117662526          2064972334   
6144        6144            117679048          2064988856   
...          ...                  ...                 ...   
8104        8104            150295813          2097605621   
8105        8105            150312224          2097622032   
8106        8106            150328865          2097638673   
8107        8107            150345732          2097655540   
8109        8109            150362146          2097671954   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
6140          2.080195e+09           NaN           NaN                NaN   
6141          2.080212e+09         955.0         288.0              910.0   
6142          2.080229e+09         840.0         272.0              812.0   
6143          2.080245e+09           NaN           NaN                NaN   
6144          2.080262e+09         673.0         231.0              638.0   
...                    ...           ...           ...                ...   
8104          2.112876e+09         288.0         505.0              266.0   
8105          2.112893e+09         316.0         547.0                NaN   
8106          2.112909e+09         306.0         507.0              283.0   
8107          2.112926e+09         294.0         504.0              291.0   
8109          2.112943e+09         303.0         542.0                NaN   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
6140                NaN                 NaN                 NaN  ...   
6141              283.0              1000.0               292.0  ...   
6142              262.0               867.0               282.0  ...   
6143                NaN                 NaN                 NaN  ...   
6144              218.0               709.0               243.0  ...   
...                 ...                 ...                 ...  ...   
8104              470.0               310.0               540.0  ...   
8105                NaN               316.0               547.0  ...   
8106              476.0               328.0               538.0  ...   
8107              475.0               297.0               534.0  ...   
8109                NaN               303.0               542.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
6140                          NaN                           NaN   
6141                       0.2621                        0.5211   
6142                       0.2429                        0.4515   
6143                          NaN                           NaN   
6144                       0.2017                        0.3692   
...                           ...                           ...   
8104                       0.4352                        0.1613   
8105                          NaN                        0.1647   
8106                       0.4409                        0.1711   
8107                       0.4394                        0.1548   
8109                          NaN                        0.1577   

      Gaze point right Y (MCSnorm)  Eye movement type  \
6140                           NaN       EyesNotFound   
6141                        0.2707            Saccade   
6142                        0.2613            Saccade   
6143                           NaN       EyesNotFound   
6144                        0.2252           Fixation   
...                            ...                ...   
8104                        0.4998           Fixation   
8105                        0.5065           Fixation   
8106                        0.4977           Fixation   
8107                        0.4942           Fixation   
8109                        0.5015           Fixati

In [31]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-8/user-8-pic-4.csv")

### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-8`|
|Recording name| 録画名|`Recording9`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:01:24.322`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:01:24.322`|
|Recording duration| 録画時間|`194848`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`9.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.94`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`34`pixels|
|Average validation accuracy | 平均バリデーション精度|`13.3`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.39`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.34`degrees|
|Average validation accuracy |平均バリデーション精度| `46`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `11`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [32]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
2026        2026             49036400          1996346208   
2027        2027             49052718          1996362526   
2028        2028             49069451          1996379259   
2029        2029             49086317          1996396125   
2030        2030             49102475          1996412283   
...          ...                  ...                 ...   
3957        3957             81169694          2028479502   
3958        3958             81186579          2028496387   
3960        3960             81203348          2028513156   
3961        3961             81219567          2028529375   
3962        3962             81236498          2028546306   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
2026          2.011623e+09         631.0         477.0              583.0   
2027          2.011640e+09           NaN           NaN                NaN   
2028          2.011656e+09         620.0         481.0              620.0   
2029          2.011673e+09         654.0         473.0              630.0   
2030          2.011689e+09           NaN           NaN                NaN   
...                    ...           ...           ...                ...   
3957          2.043755e+09        1127.0         244.0             1104.0   
3958          2.043771e+09        1126.0         253.0             1105.0   
3960          2.043788e+09           NaN           NaN                NaN   
3961          2.043804e+09        1128.0         248.0             1110.0   
3962          2.043821e+09        1127.0         251.0             1112.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
2026              474.0               680.0               479.0  ...   
2027                NaN                 NaN                 NaN  ...   
2028              481.0                 NaN                 NaN  ...   
2029              483.0               678.0               464.0  ...   
2030                NaN                 NaN                 NaN  ...   
...                 ...                 ...                 ...  ...   
3957              256.0              1150.0               232.0  ...   
3958              262.0              1146.0               245.0  ...   
3960                NaN                 NaN                 NaN  ...   
3961              257.0              1147.0               240.0  ...   
3962              258.0              1142.0               244.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
2026                       0.4388                        0.3541   
2027                          NaN                           NaN   
2028                       0.4455                           NaN   
2029                       0.4471                        0.3529   
2030                          NaN                           NaN   
...                           ...                           ...   
3957                       0.2366                        0.5989   
3958                       0.2422                        0.5970   
3960                          NaN                           NaN   
3961                       0.2377                        0.5972   
3962                       0.2387                        0.5945   

      Gaze point right Y (MCSnorm)  Eye movement type  \
2026                        0.4436       Unclassified   
2027                           NaN       EyesNotFound   
2028                           NaN            Saccade   
2029                        0.4296            Saccade   
2030                           NaN       EyesNotFound   
...                            ...                ...   
3957                        0.2147           Fixation   
3958                        0.2268           Fixation   
3960                           NaN           Fixation   
3961                        0.2224           Fixation   
3962                        0.2257           Fixati

In [33]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-8/user-8-pic-2.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-8`|
|Recording name| 録画名|`Recording9`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`17:01:24.322`|
|Recording start time UTC| 録画開始時刻(UTC) |`08:01:24.322`|
|Recording duration| 録画時間|`194848`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`9.7`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`0.94`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`34`pixels|
|Average validation accuracy | 平均バリデーション精度|`13.3`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.5`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`3.2`mm|
|Average validation accuracy |平均バリデーション精度| `1.39`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.27`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.34`degrees|
|Average validation accuracy |平均バリデーション精度| `46`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `11`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [34]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
8405         8405            155398157          2102707965   
8406         8406            155415024          2102724832   
8407         8407            155432185          2102741993   
8408         8408            155448127          2102757935   
8409         8409            155464942          2102774750   
...           ...                  ...                 ...   
10459       10459            189579439          2136889247   
10460       10460            189595504          2136905312   
10461       10461            189612394          2136922202   
10463       10463            189629362          2136939170   
10464       10464            189645474          2136955282   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
8405           2.117978e+09         944.0         617.0              954.0   
8406           2.117995e+09         930.0         607.0              929.0   
8407           2.118012e+09         937.0         602.0              937.0   
8408           2.118028e+09         933.0         604.0              935.0   
8409           2.118045e+09         941.0         607.0              938.0   
...                     ...           ...           ...                ...   
10459          2.152158e+09           NaN           NaN                NaN   
10460          2.152174e+09        1094.0         764.0             1094.0   
10461          2.152191e+09        1099.0         727.0                NaN   
10463          2.152208e+09           NaN           NaN                NaN   
10464          2.152224e+09        1071.0         741.0             1098.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
8405               623.0               934.0               612.0  ...   
8406               605.0               930.0               609.0  ...   
8407               602.0                 NaN                 NaN  ...   
8408               600.0               931.0               608.0  ...   
8409               610.0               945.0               603.0  ...   
...                  ...                 ...                 ...  ...   
10459                NaN                 NaN                 NaN  ...   
10460              764.0                 NaN                 NaN  ...   
10461                NaN              1099.0               727.0  ...   
10463                NaN                 NaN                 NaN  ...   
10464              764.0              1044.0               718.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
8405                        0.5771                        0.4863   
8406                        0.5602                        0.4846   
8407                        0.5571                           NaN   
8408                        0.5554                        0.4851   
8409                        0.5652                        0.4923   
...                            ...                           ...   
10459                          NaN                           NaN   
10460                       0.7075                           NaN   
10461                          NaN                        0.5723   
10463                          NaN                           NaN   
10464                       0.7072                        0.5436   

       Gaze point right Y (MCSnorm)  Eye movement type  \
8405                         0.5664           Fixation   
8406                         0.5638           Fixation   
8407                            NaN           Fixation   
8408                         0.5626           Fixation   
8409                         0.5581           Fixation   
...                             ...                ...   
10459                           NaN       EyesNotFound   
10460                           NaN            Saccade   
10461                        0.6735            Saccade   
10463                           NaN       EyesNotFo

In [35]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-8/user-8-pic-3.csv")